devito/passes/clusters/implicit.py

```diff
@@ -106,6 +107,13 @@ def callback(self, clusters, prefix):
             # The "implicit expressions" created for the MultiSubDomain
             exprs, dims, sub_iterators = make_implicit_exprs(d.msd, c)
 
+            # Make sure the "implicit expressions" aren't scheduled in
+            # an inner loop. E.g schedule both for `t, xi, yi` and `t, d, xi, yi`
+            edims = set(retrieve_dimensions(exprs, deep=True))
+            if dim not in edims and any(d.dim in edims for d in prefix):
```

Comment:
Actually, `dim` is in prefix, so the following should suffice:

```
if any(d.dim in edims for d in prefix):
```

In fact, the following might work as well:

```
if prefix.itdims & edims:
```